src/app/vm/shared/vm-snapshots-checker-dialog/vm-snapshots-checker-dialog.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, Inject, OnDestroy, OnInit, Type } from '@angular/core';
+import { MAT_DIALOG_DATA } from '@angular/material';
+import { Subscription } from 'rxjs';
+import { VmSnapshotService } from '../../../shared/services/vm-snapshot.service';
+import { VirtualMachine } from '../vm.model';
+
+export interface VmSnapshotsCheckerDialogData {
+  vm: VirtualMachine;
+  component: Type<any>;
```

Comment:
Can we use a generic here if we need component's type? If we don't, we can use unknown instead.